test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,505 @@
+}
+
+void
+cleanup_size_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_SIZE *curr = bms->sz_cache;
+    WT_SIZE *tmp;
+    while (curr != NULL) {
+        tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_and_cleanup_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    int i = 0;
+
+    REQUIRE(bms != nullptr);
+    WT_SIZE *curr = bms->sz_cache;
+
+    REQUIRE(bms->sz_cache_cnt == expected_items);
+    for (; i < expected_items; i++) {
+        REQUIRE(bms->sz_cache != nullptr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+    cleanup_size_list(bms);
+}
+
+TEST_CASE("Block session: __block_ext_alloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_EXT *ext;
+
+    ext = nullptr;
+    __wt_random_init(&session->getWtSessionImpl()->rnd);
+
+    REQUIRE(__ut_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+    validate_and_cleanup_ext_block(ext);
+}
+
+TEST_CASE("Block session: __block_ext_prealloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate zero extent blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_and_cleanup_ext_list(bms, 0);
+    }
+
+    SECTION("Allocate one ext blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 1) == 0);
+        validate_and_cleanup_ext_list(bms, 1);
+    }
+
+    SECTION("Allocate multiple ext blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_and_cleanup_ext_list(bms, 3);
+    }
+}
+
+TEST_CASE("Block session: __block_size_alloc", "[block_session]")
+{
+    WT_SIZE *sz;
+
+    sz = nullptr;
+    REQUIRE(__ut_block_size_alloc(nullptr, &sz) == 0);
+    validate_and_cleanup_size_block(sz);
+}
+
+TEST_CASE("Block session: __block_size_prealloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate zero size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_and_cleanup_size_list(bms, 0);
+    }
+
+    SECTION("Allocate one size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 1) == 0);
+        validate_and_cleanup_size_list(bms, 1);
+    }
+
+    SECTION("Allocate multiple size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_and_cleanup_size_list(bms, 3);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_ext_alloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
```

Comment:
I would move these declarations below the first `SECTION` otherwise it is a bit confusing for the reader.

Reply:
Done